metricbeat/helper/xpack/xpack.go

```diff
@@ -0,0 +1,47 @@
+package xpack
+
+import (
+	"fmt"
+	"time"
+)
+
+// Product supported by X-Pack Monitoring
+type Product int
+
+const (
+	// Elasticsearch product
+	Elasticsearch Product = iota
+
+	// Kibana product
+	Kibana
+
+	// Logstash product
+	Logstash
+
+	// Beats product
+	Beats
+)
+
+func (p Product) String() string {
+	indexProductNames := []string{
+		"es",
+		"kibana",
+		"logstash",
+		"beats",
+	}
+
+	if int(p) < 0 || int(p) > len(indexProductNames) {
+		panic("Unknown product")
+	}
+
+	return indexProductNames[p]
+}
+
+// MakeMonitoringIndexName method returns the name of the monitoring index for
+// a given product { elasticsearch, kibana, logstash, beats }
+func MakeMonitoringIndexName(product Product) string {
+	today := time.Now().UTC().Format("2006.01.02")
+	const version = "6"
```

Comment:
Is there some way to look up metricbeat's major version at runtime? Then we can remove this hardcoding here. Otherwise we'll have to remember to update it manually every major release.

Reply:
It is part of the `beat.Info`type but it's not global :-( Let's hard code it for now but add a TODO / note here.